### PR TITLE
Implement statistics endpoint

### PR DESCRIPTION
**Pull request by Thor77**
[TURNIERE-183](https://jiratinf16b4.it.dh-karlsruhe.de:8443/browse/TURNIERE-183)

### CODE REVIEW

File: spec/services/statistics_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+RSpec.describe StatisticsService do
+  before do
+    # build tournament with predictable test data
+    tournament = create(:tournament)
+    group_stage = create(:group_stage)
+    group = group_stage.groups.first
+    @most_dominant_score = GroupScore.new team: create(:team),
+                                          group_points: 100,
+                                          scored_points: 100, received_points: 0
+    @least_dominant_score = GroupScore.new team: create(:team),
```
Comment (Malaber) [inline spec/services/statistics_service_spec.rb]: I think you should use a factory here

File: app/models/stage.rb [hunk 2]
```diff
@@ -8,8 +8,12 @@ class Stage < ApplicationRecord
...
-    return matches.map(&:teams).flatten.uniq unless matches.size.zero?
-
-    groups.map(&:teams).flatten.uniq unless groups.size.zero?
+    if !matches.size.zero?
+      matches.map(&:teams).flatten.uniq
+    elsif !groups.size.zero?
+      groups.map(&:teams).flatten.uniq
+    else
+      []
+    end
```
Comment (Malaber) [inline app/models/stage.rb:17]: Didn't we already do that change? :thinking:

File: spec/controllers/statistics_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        get :index, params: { tournament_id: create(:tournament).to_param }
+        expect(response).to have_http_status(:not_implemented)
+      end
+    end
+
+    context 'tournament with a group stage' do
+      before do
+        @tournament = create(:group_stage_tournament)
+        @group_stage = @tournament.stages.find_by(level: -1)
+        @most_dominant_score = GroupScore.new team: @tournament.teams.first,
```
Comment (Malaber) [inline spec/controllers/statistics_controller_spec.rb]: Again, I think a factory would be great here

**Review by Malaber — CHANGES_REQUESTED** [review]

File: app/models/stage.rb (hunk 2 above)
Comment (Thor77) [reply app/models/stage.rb:17]: Considering the commit is changing something, it doesn't look like it.

File: spec/services/statistics_service_spec.rb (hunk 1 above)
Comment (Thor77) [reply spec/services/statistics_service_spec.rb]: How should I name it? And what should it contain?
Imo it's not worth the overhead, but if you think so, I will oblige.

Comment (Malaber) [reply spec/services/statistics_service_spec.rb]: `:group_score` ?

Comment (Malaber) [reply spec/services/statistics_service_spec.rb]: It think it is good just to separate test from test data where possible..

File: app/models/stage.rb (hunk 2 above)
Comment (Malaber) [reply app/models/stage.rb:17]: well yes :D

**Review by Malaber — APPROVED** [review]
lgtm

**Pull request merged**